datapipelines/src/main/kotlin/io/customer/sdk/android/CustomerIO.kt

```diff
@@ -157,11 +186,70 @@ class CustomerIO private constructor(
     }
 
     override fun clearIdentify() {
-        val userId = registeredUserId ?: "anonymous"
-        logger.debug("resetting user profile with id $userId")
+        logger.info("resetting user profile with id $registeredUserId")
+
+        logger.debug("deleting device token to remove device from user profile")
+        deleteDeviceToken()
+
+        logger.debug("resetting user profile")
         analytics.reset()
     }
 
+    override val registeredDeviceToken: String?
+        get() = globalPreferenceStore.getDeviceToken()
+
+    override var deviceAttributes: CustomAttributes
+        get() = emptyMap()
+        set(value) {
+            trackDeviceAttributes(registeredDeviceToken, value)
+        }
+
+    override fun registerDeviceToken(deviceToken: String) {
+        if (deviceToken.isBlank()) {
+            logger.debug("device token cannot be blank. ignoring request to register device token")
+            return
+        }
+
+        logger.info("registering device token $deviceToken for user profile: $registeredUserId")
+
+        logger.debug("storing device token to device storage")
+        globalPreferenceStore.saveDeviceToken(deviceToken)
+
+        trackDeviceAttributes(deviceToken)
+    }
+
+    private fun trackDeviceAttributes(token: String?, attributes: CustomAttributes = emptyMap()) {
+        if (token.isNullOrBlank()) {
+            logger.debug("no device token found. ignoring request to track device.")
+            return
+        }
+
+        val existingDeviceToken = contextPlugin.deviceToken
+        if (existingDeviceToken != null && existingDeviceToken != token) {
+            logger.debug("token has been refreshed, deleting old token to avoid registering same device multiple times")
+            deleteDeviceToken()
+        }
+
+        // TODO: Append auto tracked device attributes here
```

Comment:
This will be covered in [MBL-215](https://linear.app/customerio/issue/MBL-215/automatic-device-attributes-collection-and-user-agent-updates)